src/character.cpp

```diff
@@ -8383,6 +8391,53 @@ void Character::update_stamina( int turns )
     set_stamina( std::min( std::max( get_stamina(), 0 ), max_stam ) );
 }
 
+int Character::get_cardio() const
+{
+    return cardio;
+}
+
+void Character::set_cardio( int new_cardio )
+{
+    cardio = new_cardio;
+}
+
+void Character::update_cardio()
+{
+    const int bmr = get_bmr();
+    // health scaling constant, value tbd
+    constexpr float health_mult = 2;
+    // skill scaling constant, value tbd
+    constexpr float skill_mult = 16;
+    // THIS WILL BE SKILL_ATHLETICS IN THE NEAR FUTURE
+    const int athletics_mod = get_skill_level( skill_swimming ) * skill_mult;
+    const int health_effect = get_healthy() * health_mult;
+    // Traits now exclusively affect cardio, NOT max_stamina directly. In the future, make cardio_acc also be affected by cardio traits so that they don't become less impactful.
+    // TODO: Rename max_stamina_modifier
+    static const std::string max_stamina_modifier( "max_stamina_modifier" );
+    // with no traits : trait_mod = 0. With indefatigable: trait_mod = 64 (40% of what maximum athletics skill gives you). With the bad cardio trait: trait_mod = -80 (too severe??).
+    const int trait_mod = std::round( ( mutation_value( max_stamina_modifier ) - 1.0f ) * 320 );
+    //WIP. Currently no proficiencies seem suitable as of build 11051
+    const int prof_mod = 0;
+    const int cardio_acc_mod = get_cardio_acc();
+    set_cardio( bmr / 2 + athletics_mod + health_effect + trait_mod + prof_mod + cardio_acc_mod );
+}
+
+int Character::get_cardio_acc() const
+{
+    return cardio_acc;
+}
+
+void Character::set_cardio_acc( int new_cardio_acc )
+{
+    cardio_acc = new_cardio_acc;
+}
+
+void Character::update_cardio_acc()
+{
+    //WIP- LATER PR
+    cardio_acc = 0;
+}
```

Comment:
maybe dont' include this until said later pr

Reply:
If cardio_acc is 0, that leaves cardio based on half of BMR only.